Sources/GRPC/ClientCalls/BidirectionalStreamingCall.swift

```diff
@@ -14,24 +14,106 @@
+  }
+
+  /// The `EventLoop` this call is running on.
+  public var eventLoop: EventLoop {
+    return self.transport.eventLoop
+  }
+
+  /// Cancel this RPC if it hasn't already completed.
+  public func cancel(promise: EventLoopPromise<Void>?) {
+    self.transport.cancel(promise: promise)
+  }
+
+  // MARK: - Response Parts
+
+  /// The initial metadata returned from the server.
+  public let initialMetadata: EventLoopFuture<HPACKHeaders>
+
+  /// The trailing metadata returned from the server.
+  public let trailingMetadata: EventLoopFuture<HPACKHeaders>
+
+  /// The final status of the the RPC.
+  public let status: EventLoopFuture<GRPCStatus>
+
+  // MARK: - Requests
+
+  /// Sends a message to the service.
+  ///
+  /// - Important: Callers must terminate the stream of messages by calling `sendEnd()` or
+  ///   `sendEnd(promise:)`.
+  ///
+  /// - Parameters:
+  ///   - message: The message to send.
+  ///   - compression: Whether compression should be used for this message. Ignored if compression
+  ///     was not enabled for the RPC.
+  ///   - promise: A promise to fulfill with the outcome of the send operation.
+  public func sendMessage(
+    _ message: RequestPayload,
+    compression: Compression = .deferToCallDefault,
+    promise: EventLoopPromise<Void>?
+  ) {
+    let compressed = compression.isEnabled(enabledOnCall: self.options.messageEncoding.enabledForRequests)
+    let messageContext = _MessageContext(message, compressed: compressed)
+    self.transport.sendRequest(.message(messageContext), promise: promise)
+  }
+
+  /// Sends a sequence of messages to the service.
+  ///
+  /// - Important: Callers must terminate the stream of messages by calling `sendEnd()` or
+  ///   `sendEnd(promise:)`.
+  ///
+  /// - Parameters:
+  ///   - messages: The sequence of messages to send.
+  ///   - compression: Whether compression should be used for this message. Ignored if compression
+  ///     was not enabled for the RPC.
+  ///   - promise: A promise to fulfill with the outcome of the send operation. It will only succeed
+  ///     if all messages were written successfully.
+  public func sendMessages<S>(
+    _ messages: S,
+    compression: Compression = .deferToCallDefault,
+    promise: EventLoopPromise<Void>?
+  ) where S: Sequence, S.Element == RequestPayload {
+    let compressed = compression.isEnabled(enabledOnCall: self.options.messageEncoding.enabledForRequests)
+    self.transport.sendRequests(messages.map {
+      .message(_MessageContext($0, compressed: compressed))
+    }, promise: promise)
+  }
+
+  /// Terminates a stream of messages sent to the service.
+  ///
+  /// - Important: This should only ever be called once.
+  /// - Parameter promise: A promise to be fulfilled when the end has been sent.
+  public func sendEnd(promise: EventLoopPromise<Void>?) {
+    self.transport.sendRequest(.end, promise: promise)
+  }
+
+  public func newMessageQueue() -> EventLoopFuture<Void> {
```

Comment:
It's a requirement on the `ClientCall` protocol; I think it was added right at the start as a way to chain sending messages: `rpc.newMessageQueue().flatMap { rpc.sendMessage(...) }` etc., I don't think we've reconsidered its existence though since it clearly doesn't make a lot of sense!